src/lava/utils/profiler.py

```diff
@@ -2,13 +2,40 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # See: https://spdx.org/licenses/
 
-"""
-This module will contain a tool to determine power and performance of workloads
-for Loihi 1 or Loihi 2 based on software simulations or hardware measurements.
-
-The execution time and energy of a workload will be either measured on hardware
-during execution or estimated in simulation. The estimation is based on
-elementary hardware operations which are counted during the simulation. Each
-elementary operation has a defined execution time and energy cost, which is
-used in a performance model to calculate execution time and energy.
-"""
+import warnings
+import typing as ty
+from lava.magma.core.run_configs import RunConfig, Loihi2HwCfg
+
+# Dictionary to relate a certain RunCfg to specific Profiler
+run_cfg_dict: ty.Dict[RunConfig, RunConfig] = {}
+
+try:
+    from lava.utils.l2_profiler import Loihi2HWProfiler
+    run_cfg_dict[Loihi2HwCfg] = Loihi2HWProfiler
+except ModuleNotFoundError:
+    warnings.warn("Loihi2HWProfiler could not be imported. "
+                  "Currently no profiler is available.")
+
+
+class Profiler:
+    """Base class for profiling execution time, energy and other
+    metrics on different Ressources. Depending on the computing
+    ressource an appropriate profiler needs to be chosen. The run
+    configuration is used to choose the related profiler, if there
+    is one."""
+
+    @staticmethod
+    def init(run_cfg: RunConfig):
+        """Decide which profiler is needed based on the run
+        configuration."""
+
+        if not isinstance(run_cfg, RunConfig):
+            raise AssertionError("<run_cfg> must be an "
```

Comment:
I would prefer a TypeError here, since AssertionError to me implies this is not a runtime check but a debugging tool, which is not the case here. But this is a quibble really, probably doesn't matter.

Reply:
Yes that does make sense. Changed it to TypeError.